sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,280 @@
+      processor = app.get(FullTimeMSFAAProcessResponseIntegrationScheduler);
+    });
+
+    beforeEach(async () => {
+      jest.clearAllMocks();
+      // Force any not signed MSFAA to be signed to ensure that new
+      // ones created will be the only ones available to be updated.
+      await db.msfaaNumber.update(
+        { dateSigned: IsNull() },
+        { dateSigned: getISODateOnlyString(new Date()) },
+      );
+      // Cancel any not canceled MSFAA.
+      await db.msfaaNumber.update(
+        { cancelledDate: IsNull() },
+        { cancelledDate: getISODateOnlyString(new Date()) },
+      );
+    });
+
+    it("Should reactivate a cancelled MSFAA when the same MSFAA is received in the response file.", async () => {
+      // Arrange
+      const student = await saveFakeStudent(db.dataSource);
+      const msfaaInputData = [MSFAA_FULL_TIME_MARRIED];
+      const createdMSFAARecords = await saveMSFAATestInputsData(
+        db,
+        msfaaInputData,
+        { student },
+        { state: MSFAAStates.CancelledOtherProvince },
+      );
+      // Queued job.
+      const job = createMock<Job<void>>();
+      mockDownloadFiles(sftpClientMock, [
+        MSFAA_FULL_TIME_RECEIVE_FILE_WITH_REACTIVATION_RECORD,
+      ]);
+      // Act
+      const processResult = await processor.processMSFAA(job);
+      // Assert
+      expect(processResult).toStrictEqual([
+        {
+          processSummary: [
+            `Processing file ${MSFAA_FULL_TIME_RECEIVE_FILE_WITH_REACTIVATION_RECORD}.`,
+            "File contains:",
+            "Confirmed MSFAA records (type R): 1.",
+            "Cancelled MSFAA records (type C): 0.",
+            "Record from line 1, updated as confirmed.",
+          ],
+          errorsSummary: [],
+        },
+      ]);
+      // Assert that the file was deleted from SFTP.
+      expect(sftpClientMock.delete).toHaveBeenCalled();
+      // Find the updated MSFAA records previously created.
+      const msfaaIDs = createdMSFAARecords.map((msfaa) => msfaa.id);
+      const msfaaUpdatedRecords = await db.msfaaNumber.find({
+        select: {
+          msfaaNumber: true,
+          dateSigned: true,
+          serviceProviderReceivedDate: true,
+          cancelledDate: true,
+          newIssuingProvince: true,
+        },
+        where: {
+          id: In(msfaaIDs),
+        },
+        order: {
+          msfaaNumber: "ASC",
+        },
+      });
+      expect(msfaaUpdatedRecords).toHaveLength(msfaaInputData.length);
+      const [reactivatedMSFAA] = msfaaUpdatedRecords;
+      // Validate reactivated confirmed record.
+      expect(reactivatedMSFAA.dateSigned).toBe("2023-05-02");
+      expect(reactivatedMSFAA.serviceProviderReceivedDate).toBe("2023-05-03");
+      expect(reactivatedMSFAA.cancelledDate).toBe(null);
+    });
+
+    it("Should reactivate a cancelled MSFAA when the same MSFAA is received in the response file and re-associate this reactivated MSFAA with all pending disbursements.", async () => {
```

Comment:
do we need test 69? because this is testing the same right + reassociate?

Reply:
@ann-aot this additionally validates if any pending disbursements which is attached to an MSFAA which got cancelled because of the re-activation of MSFAAOld are now associated to MSFAAOld.

Reply:
the test case on line 69 doesn't help with any additional coverage and basically is a subset of the testcase on line 148. Since there is no additional code coverage with the smaller test case, removing it.